### ISSUE TITLE
bug report, why new_x = candidates[0].detach()?

### ISSUE THREAD

**Issue by machanic** [satyanshukla/bayes_attack#3]
In https://github.com/satyanshukla/bayes_attack/blob/main/attack.py#L102
The code of `new_x = candidates[0].detach()` would cause transform error!
because shape of `candidates` is `[1, 1944]`.
Because the `pert` does not have axis=1(dim=1) in the statement of `t_dim = int(np.sqrt(pert.shape[1]/(2*channel)))`.  (https://github.com/satyanshukla/bayes_attack/blob/main/utils.py#L70)

**Comment by satyanshukla** [comment]
Thanks for pointing this out. I have fixed this in the new commit.
